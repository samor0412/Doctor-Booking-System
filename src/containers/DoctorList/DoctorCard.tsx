import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import { primary, secondary } from 'colors';
import { Doctor } from 'api/getDoctor';

const HoverMask = styled.div`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  display: none;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  color: ${primary};
  font-weight: bold;
`;

const Card = styled(NavLink)`
  position: relative;
  border: 0.2rem solid ${secondary};
  border-radius: 0.4rem;
  box-shadow: 3px 6px 9px -4px rgba(0, 0, 0, 0.72);
  flex: 0 0 30%;
  box-sizing: border-box;

  &:hover ${HoverMask} {
    display: flex;
    background-color: rgba(255, 255, 255, 0.8);
  }
`;

const Name = styled.div`
  font-size: 1.2rem;
  text-align: center;
  padding: 0.6rem;
`;

const ProfileImg = styled.img`
  width: 100%;
`;

interface Props {
  doctor: Doctor;
}
const DoctorCard: React.FC<Props> = ({ doctor }) => (
  <Card to="/doctor-profile">
    <ProfileImg src={'defaultDoctorProfileImage.svg'}></ProfileImg>
    <Name>{doctor.name}</Name>
    <HoverMask>More</HoverMask>
  </Card>
);

export default DoctorCard;
