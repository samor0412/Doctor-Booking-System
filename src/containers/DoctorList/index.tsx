import React from 'react';
import styled from 'styled-components';
import { useQuery } from 'react-query';
import { Doctor } from 'api/types';
import DoctorCard from './DoctorCard';
import { HEADER_HEIGHT } from 'components/Header';
import LoadingOverlay from 'components/Loading';
import getDoctors from 'api/getDoctors';
import { Error } from 'styles';

const Container = styled.div`
  padding-top: ${HEADER_HEIGHT};
  width: 100vw;
`;

const SearchContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin: 4rem;
  gap: 4rem 3.33%;
`;

const DoctorList = () => {
  const { isLoading, isError, data: doctors } = useQuery<Doctor[]>('doctors', getDoctors);

  return (
    <Container>
      <LoadingOverlay isLoading={isLoading}>
        {isError ? (
          <Error>There is an error. Please refresh the page</Error>
        ) : (
          <SearchContainer>
            {doctors && doctors.map((doctor) => <DoctorCard key={doctor.id} doctor={doctor} />)}
          </SearchContainer>
        )}
      </LoadingOverlay>
    </Container>
  );
};

export default DoctorList;
