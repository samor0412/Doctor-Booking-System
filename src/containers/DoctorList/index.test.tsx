import React, { ReactElement } from 'react';
import { render as rtlRender, screen, waitFor } from '@testing-library/react';
import DoctorList from './index';
import { QueryClient, QueryClientProvider } from 'react-query';
import mockDoctors from './__mocks__/mockDoctors.json';
import { BrowserRouter } from 'react-router-dom';

jest.mock('api/getDoctors', () => async () => await mockDoctors);

const queryClient = new QueryClient();
const render = (element: ReactElement) =>
  rtlRender(
    <BrowserRouter>
      <QueryClientProvider client={queryClient}>{element}</QueryClientProvider>
    </BrowserRouter>
  );

describe('DoctorList', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<DoctorList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a list of doctors', async () => {
    render(<DoctorList />);
    await waitFor(() => {
      expect(screen.queryByText('KWOK KWAN MAN')).toBeInTheDocument();
    });
  });
});
