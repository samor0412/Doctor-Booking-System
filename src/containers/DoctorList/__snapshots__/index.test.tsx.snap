// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorList snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-papXJ cjaQRW"
  >
    <div
      class="sc-ftvSup jETrpz"
    >
      <img
        class="sc-jSMfEi hrLfVF"
        src="loader.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
